lib/ami_spec/aws_key_pair.rb

```diff
@@ -0,0 +1,36 @@
+require 'aws-sdk-ec2'
+require 'logger'
+require 'securerandom'
+require 'tempfile'
+require 'pathname'
+
+module AmiSpec
+  class AwsKeyPair
+
+    def self.create(**args)
+      new(**args).tap(&:create)
+    end
+
+    def initialize(ec2: Aws::EC2::Resource.new, key_name_prefix: 'ami-spec-', logger: Logger.new(STDOUT))
+      @ec2 = ec2
+      @key_name = "#{key_name_prefix}#{SecureRandom.uuid}"
+      @logger = logger
+    end
+
+    attr_reader :key_name, :key_file
+
+    def create
+      @logger.info "Creating temporary AWS key pair: #{@key_name}"
+      @key_pair = @ec2.create_key_pair(key_name: @key_name)
+      @temp_file = Tempfile.new('key')
+      @temp_file.write(@key_pair.key_material)
+      @temp_file.close
+      @key_file = Pathname.new(@temp_file.path)
+    end
+
+    def delete
+      @logger.info "Deleting temporary AWS key pair: #{@key_name}"
+      @key_pair.delete
```

Comment:
Does this delete it in AWS?

Reply:
Yes, exactly. It gets delegated to the client ([code](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-ec2/lib/aws-sdk-ec2/key_pair.rb#L173-L189)).